File: src/types/aggregation.ts
export interface AggregationResult {
  tickets_by_status: CountEntry[];
  tickets_by_priority: CountEntry[];
  tickets_by_category: CountEntry[];
  tickets_over_time: TimeSeriesEntry[];
  resolution_time_by_priority: AvgEntry[];
  summary: SummaryStats;
}

export interface CountEntry {
  name: string;
  count: number;
}

export interface TimeSeriesEntry {
  date: string;
  created: number;
  resolved: number;
}

export interface AvgEntry {
  name: string;
  avg_hours: number;
  median_hours: number;
  count: number;
}

export interface SummaryStats {
  total_tickets: number;
  open_tickets: number;
  resolved_tickets: number;
  avg_resolution_hours: number;
  median_resolution_hours: number;
}

File: src-tauri/src/models/aggregation.rs
use serde::Serialize;

#[derive(Serialize)]
pub struct AggregationResult {
    pub tickets_by_status: Vec<CountEntry>,
    pub tickets_by_priority: Vec<CountEntry>,
    pub tickets_by_category: Vec<CountEntry>,
    pub tickets_over_time: Vec<TimeSeriesEntry>,
    pub resolution_time_by_priority: Vec<AvgEntry>,
    pub summary: SummaryStats,
}

#[derive(Serialize)]
pub struct CountEntry {
    pub name: String,
    pub count: u32,
}

#[derive(Serialize)]
pub struct TimeSeriesEntry {
    pub date: String, // "2025-01" (month) or "2025-W03" (week)
    pub created: u32,
    pub resolved: u32,
}

#[derive(Serialize)]
pub struct AvgEntry {
    pub name: String,
    pub avg_hours: f64,
    pub median_hours: f64,
    pub count: u32,
}

#[derive(Serialize)]
pub struct SummaryStats {
    pub total_tickets: u32,
    pub open_tickets: u32,
    pub resolved_tickets: u32,
    pub avg_resolution_hours: f64,
    pub median_resolution_hours: f64,
}
